### ISSUE TITLE
New Method Too Slow

### ISSUE THREAD

**Issue by XXXXXXXXXIII** [xuzhengyi1995/Manga_downloader#87]
Hi, the performance for the new method is very slow when using bookwalker reader, the page would get stuck for half a minute at a time and is completely unresponsive to user input. Is this a known issue? Can it be improved somehow? 

Thanks~

**Comment by XXXXXXXXXIII** [comment]
I'm not sure how exactly the custom chrome was implemented, but it seems like `toDataURL()` runs much much slower than `toBlob()` when extracting image from canvas.

**Comment by xuzhengyi1995** [comment]
The `toDataURL()` and `toBlob()` is calling the same function [ImageDataBuffer::EncodeImageInternal](https://chromium.googlesource.com/chromium/src/+/d039b77b3c363d2e544a3c4d95cfa3c87d839b1b/third_party/blink/renderer/platform/graphics/image_data_buffer.cc#142) when encoding the image, the different thing is `toDataURL()`  is adding a base64 encode, but because we are modifying the code in C++ so we also called this function to encode the image.

I think the things made it slow is recording every step of the drawing and also calculate the hash to make sure we do not save the same page multiple times, there are many implementation much better and could also make this feature much more convenient, but it really need much more time.

PS: I have also tried this on some computers before but it seems not too slow and even be feel no difference with the normal browser.
